### PR TITLE
Array value format is added

### PR DESCRIPTION
**Pull request by kainobor**

### CODE REVIEW

File: lib/cimp/kv.go [hunk 1]
```diff
@@ -3,141 +3,87 @@ package cimp
...
-func ToSnakeCase(str string) string {
-	str = matchAllSpecSymbols.ReplaceAllString(str, "_")
-	str = matchFirstCap.ReplaceAllString(str, "${1}_${2}")
-	str = matchAllCap.ReplaceAllString(str, "${1}_${2}")
-	str = matchAllMultipleUnderscore.ReplaceAllString(str, "_")
-	str = strings.Trim(str, "_")
-
-	return strings.ToLower(str)
+func (kv *KV) AddPrefix(prefix string) {
+	if len(prefix) > 0 && prefix[len(prefix)-1] != '/' {
```
Comment (unkeep) [inline lib/cimp/kv.go]: if !strings.HasSuffix(prefix, "/") ...

File: lib/cimp/storage.go [hunk 2]
```diff
@@ -28,46 +30,43 @@ func NewStorage(cfg Config) (*ConsulStorage, error) {
...
+			return fmt.Errorf("get key %q value from tree: %w", key, err)
+		}
+
 		op := &api.TxnOp{
 			KV: &api.KVTxnOp{
 				Verb:  api.KVSet,
-				Key:   k,
-				Value: []byte(fmt.Sprintf("%v", v)),
+				Key:   kv.globalPrefix + key,
+				Value: []byte(fmt.Sprintf("%v", leaf.Value)),
```
Comment (unkeep) [inline lib/cimp/storage.go]: fmt.Sprint(leaf.Value) ?

File: lib/cimp/storage.go [hunk 3]
```diff
@@ -28,46 +30,43 @@ func NewStorage(cfg Config) (*ConsulStorage, error) {
...
+			if ok, _, _, err := cs.client.Txn().Txn(ops, nil); !ok {
+				return fmt.Errorf("execute consul SET-transaction: %w", err)
+			}
+			ops = nil
+		}
 	}
 
-	ok, _, _, err := cs.client.Txn().Txn(ops, nil)
-	if !ok {
+	if ok, _, _, err := cs.client.Txn().Txn(ops, nil); !ok {
```
Comment (unkeep) [inline lib/cimp/storage.go]: if len(ops) != 0 ...

File: lib/tree/tree.go [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+		mt.Order = append(mt.Order, name)
+	}
+	mt.Content[name] = value
+}
+
+func (mt *Tree) ShallowClone() *Tree {
+	newOrder := make([]string, len(mt.Order))
+	copy(newOrder, mt.Order)
+
+	newContent := make(map[string]Marshalable)
```
Comment (unkeep) [inline lib/tree/tree.go]: specify map capacity

**Review by unkeep — CHANGES_REQUESTED** [review]

File: lib/cimp/storage.go (hunk 2 above)
Comment (kainobor) [reply lib/cimp/storage.go]: fixed in next commit

File: lib/cimp/storage.go (hunk 3 above)
Comment (kainobor) [reply lib/cimp/storage.go]: fixed in next commit

File: lib/tree/tree.go (hunk 4 above)
Comment (kainobor) [reply lib/tree/tree.go]: fixed in next commit

**Review by unkeep — APPROVED** [review]

**Pull request merged**